<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>DIVERSITY traits : imputation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<link href="site_libs/ionicons-2.0.1/css/ionicons.min.css" rel="stylesheet" />




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Maya Guéguen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About me
  </a>
</li>
<li>
  <a href="publications.html">
    <span class="ion ion-document-text"></span>
     
    Publications
  </a>
</li>
<li>
  <a href="work.html">
    <span class="fa fa-chart-line"></span>
     
    Work
  </a>
</li>
<li>
  <a href="tutorials.html">
    <span class="fa fa-toolbox"></span>
     
    Tutorials
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="mailto:maya.gueguen@univ-grenoble-alpes.fr">
    <span class="fa fa-envelope-o"></span>
     
    Contact me
  </a>
</li>
<li>
  <a href="https://github.com/MayaGueguen">
    <span class="fa fa-github"></span>
     
    
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">DIVERSITY traits : imputation</h1>
<h4 class="date">26/10/2020</h4>

</div>


<p><br/> <br/></p>
<div id="glossary" class="section level1">
<h1>Glossary</h1>
<ul>
<li><strong>Distribution of missing data</strong> =
<ul>
<li>MCAR = missing completely at random</li>
<li>MAR = missing at random (when the presence of missing data for a given variable is related to the values of another variable)</li>
<li>MNAR = missing not at random</li>
</ul></li>
</ul>
<p><span class="math inline">\(\Rightarrow\)</span> in functional trait databases, missingness is related to the frequency of the species and their abundances, and taxonomic and phylogenetic bias is commonplace <span class="math inline">\(/!\setminus\)</span></p>
<p><br/></p>
<ul>
<li><strong>Deletion method</strong> =
<ul>
<li>delete species with missing data for the calculation of diversity indices</li>
<li>acceptable for estimation of the community-weighted mean trait value (CWM)<br />
as long as it only concerns the minor species (should not exceed 20<span class="math inline">\(\%\)</span> of the total biomass of the community)</li>
</ul></li>
</ul>
<p><br/></p>
<ul>
<li><strong>Gower + PCoA</strong> =
<ul>
<li>compute Gower distance (with missing data) and project the distance with a Principal Coordinate Analysis,<br />
the axes being then used as functional traits</li>
<li>only relevant for functional diversity indices calculated from several traits</li>
<li>trait information gets lost and only multivariate approaches can be used</li>
</ul></li>
</ul>
<p><br/></p>
<ul>
<li><p><strong>Imputation method</strong> = replace missing data with substituted values</p></li>
<li><p><strong>Simple imputation</strong> = a single value is imputed for each missing datum</p></li>
<li><p><strong>Multiple imputation</strong> = Monte Carlo technique in which the missing values are replaced by <span class="math inline">\(m &gt; 1\)</span> imputed values, combined to produce estimates, confidence intervals, missing data uncertainty</p></li>
<li><p><strong>Likelihood-based imputation</strong> = imputation and analysis are conducted simultaneously</p></li>
</ul>
<p><br/> <br/></p>
</div>
<div id="specific-traits" class="section level1">
<h1>Specific traits</h1>
<p><br/></p>
<div id="tamme-ecology" class="section level2">
<h2>2014 : Tamme (Ecology)</h2>
<p><em>Measure the predictive power of simple plant traits to estimate species’ maximum dispersal distances :</em></p>
<ul>
<li>dispersal distance is related <strong>to a dispersal syndrome</strong> (Willson 1993, Pärtel &amp; Zobel 2007, Vittoz &amp; Engler 2007), that may generally be deduced from seed morphology, but difficult to establish a direct association (Nogales 2007)</li>
<li><strong>plant height</strong> more important than seed mass in determining seed dispersal distances (Thomson et al. 2011)</li>
<li><strong>maximum dispersal distance strongly correlated with mean dispersal distance</strong> (Thomson et al. 2011) :</li>
</ul>
<p><span class="math display">\[ log_{10}(\text{ maximumDist }) = 0.795 + 0.984 * log_{10}(\text{ meanDist })\]</span></p>
<p><br/><br/></p>
<div style="text-align:center;">
<pre>depending on the set of species used and the trait data available,
different traits are important in describing dispersal distance patterns</pre>
<p><br/></p>
<p><span class="math inline">\(\Rightarrow\)</span> <strong>(dispersal syndrome + growth form + terminal velocity)</strong><br />
explain <span class="math inline">\(\sim\)</span> 60<span class="math inline">\(\%\)</span> of the variation in maximum dispersal distances</p>
<p><span class="math inline">\(\Rightarrow\)</span> <strong>(dispersal syndrome + growth form)</strong><br />
explain <span class="math inline">\(\sim\)</span> 50<span class="math inline">\(\%\)</span> of the variation in maximum dispersal distances</p>
</div>
<p>with</p>
<ul>
<li>on average, maximum dispersal distances increase from :<br />

<div style="text-align:center;">
<p>wind (no special mechanisms) &lt; ballistic &lt; ant &lt; wind (special mechanisms) &lt; animal<br />
herb &lt; shrub &lt; tree</p>
</div></li>
<li>models with terminal velocity only apply to species with wind or ballistic dispersal syndrome</li>
<li>in general, short-distance dispersal overestimated, and long-distance dispersal underestimated</li>
</ul>
<div style="text-align:center;">
<p><span class="math inline">\(\Rightarrow\)</span> <strong><code>dispeRsal</code> package</strong></p>
</div>
<p><br/><br/></p>
</div>
</div>
<div id="comparison-of-methods" class="section level1">
<h1>Comparison of methods</h1>
<p><br/></p>
<div id="taugourdeau" class="section level2">
<h2>2014 : Taugourdeau</h2>
<p><em>Effect of imputation methods on the evaluation of trait values at species level and on the subsequent calculation of functional diversity indices at community level :</em></p>
<div style="text-align:center;">
<pre>None of these methods was able to perform best for all the traits and indices</pre>
<pre>at the species level, the most accurate imputation method is not the same for all traits 
and in all cases, but dissimilarity and/or relationships method always the most accurate 
among the single imputation methods</pre>
<pre>key parameter to choose the adequate imputation method 
= the distribution of the trait value in the dataset
applying a transformation method to improve the distribution of the trait values prior to  
using a imputation method could be useful in improving the quality of the replacement</pre>
<pre>30% of the data missing = percentage of missing data as a limit 
for the utilization of these single imputation methods</pre>
</div>
<ul>
<li><strong>Average method</strong> = least accurate</li>
<li><strong>Functional proximity between species</strong> =
<ul>
<li>best method for traits with <em>unbalanced distribution</em></li>
<li>but not an appropriate choice for a study on functional distance between species,<br />
as functional distance would then be underestimated</li>
<li>less affected by the percentage of deleted data (but Gower dissimilarity cannot be calculated between two species if no trait for both species : not possible if missing data are too numerous)</li>
</ul></li>
<li><strong>Relationships between traits</strong> =
<ul>
<li>best method for traits with <em>balanced distribution</em></li>
<li>but the most sensitive to the level of missing data</li>
</ul></li>
<li><strong>MICE</strong> = more accurate than all other methods for all traits except SLA
<ul>
<li>the correction model can be adapted to the distribution of the variable, but caution with unbalanced traits</li>
</ul></li>
</ul>
<p><br/></p>
</div>
<div id="penone" class="section level2">
<h2>2014 : Penone</h2>
<p><em>Evaluate the performance of four approaches for estimating missing values in trait databases and test whether imputed datasets retain underlying allometric relationships among traits :</em></p>
<div style="text-align:center;">
<pre>adding phylogenetic information reduced the error for all approaches except kNN 
this result was stronger for missForest than for mice

whithout phylo, mice gave better results than missForest
this difference was not significant whith phylo</pre>
<pre>bias was lower when missing data were imputed rather than deleted,
especially when more than 30% of the data were missing.

with phylo, all approaches had comparable bias, 
significantly lower than bias in datasets with missing data

phylo reduced the bias for missForest but not for mice
</pre>
<pre>imputation approaches with phylogenetic variance-covariance matrix (Phylopars) 
and phylogenetic eigenvectors (missForest and mice) gave similar results

phylo did not improve estimation equally among traits,
being more important where phylogenetic signal is stronger 
and when there are no other traits with strong signal</pre>
</div>
<p><br/></p>
</div>
<div id="schrodt" class="section level2">
<h2>2015 : Schrodt</h2>
<p><em>A new approach (BHPMF) which imputes trait values based on the taxonomic hierarchy, structure within the trait matrix and trait–environment relationships at the same time as providing uncertainty estimates for each single trait prediction.</em></p>
<div style="text-align:center;">
<pre>prediction accuracy is not related to the number of entries per trait</pre>
<pre>BHPMF outperforms the species MEAN baseline in all aspects: 
RMSE and R2 of predicted versus observed entries are smaller and larger, respectively, 
for each individual trait, and trait–trait correlations are better retrieved 

aHPMF provides a concept to extrapolate from point measurements to species ranges 
accounting for intraspecific variability</pre>
<pre>explicitly taking environmental constraints into account adds little or no improvement to BHPMF</pre>
<pre>Advancing BHPMF to account for phylogenetic distances instead of taxonomy will improve 
opportunities for efficient out-of-sample predictions, at least in well-resolved clades</pre>
</div>
<p><br/></p>
</div>
<div id="poyatos" class="section level2">
<h2>2018 : Poyatos</h2>
<p><em>Assess the performance of different imputation methods to fill simulated gaps at different missingness levels in a spatially explicit plant trait dataset</em></p>
<div style="text-align:center;">
<pre>No method performed best consistently for the five studied traits, but, considering all traits 
and performance metrics, MICE informed by relevant ecological variables gave the best results.
However, at higher missingness (>30 %), species mean imputations and regression kriging 
tended to outperform MICE for some traits.</pre>
<pre>low missingness rates (10 %) = mice and kNN more accurate (NRMSE) than Mean
moderate/high missingness = mice and kNN comparable/outperformed by Mean, and specially by OrdKrig

trait covariation did not improve imputations at high missingness

results for OrdKrig, compared to mice and kNN, show that spatial structure, rather than
trait covariation, may provide more accurate trait imputations when gaps are frequent</pre>
<pre>Introducing auxiliary variables as predictors improved MICE performance substantially 
but these improvements were dependent on the specific predictor set and trait</pre>
<pre>
Using imputed or incomplete datasets did not lead to large differences 
in the studied trait relationships when missingness was < 50%
At high missingness, using imputed datasets led to comparatively larger 
departures from the relationships obtained with the complete dataset
No imputation method appeared to perform consistently better than others in preserving trait
relationships at high missingness levels and, under these conditions, using incomplete datasets
appeared to correctly reproduce the observed trait relationships in the complete dataset
</pre>
</div>
<p><br/></p>
</div>
<div id="johnson" class="section level2">
<h2>2020 : Johnson</h2>
<p><em>Evaluate the performance of approaches for handling missing values when considering biased datasets</em></p>
<div style="text-align:center;">
<pre>Rphylopars = most accurate estimate of missing values, best preserve response–trait slope

estimates of missing data were still inaccurate, even with only 5% of values missing

Under severe biases, errors were high with every approach.
complete-case analysis frequently outperform Mice imputation and, to a lesser degree, BHPMF
</pre>
<pre>If the objective of the imputation is to produce estimates of missing values, 
single imputation is considered most effective
if the objective is to model imputed values against another variable, 
the added error in the multiple imputation is advantageous</pre>
<pre>Including phylogenetic information generally improved imputation performance in every method</pre>
<pre>Including the response in the imputation :
- when response–trait slope positive : decreased imputation error in all approaches, 
decreased slope error substantially in Mice (-> almost comparable with Rphylopars), 
increased slope error in Rphylopars and BHPMF
- when no relationship between trait and response : 
increased imputation and slope errors in every approach, but with a small effect

-> broadly advisable to include response with Mice, to exclude it with Rphylopars and BHPMF</pre>
<pre>Missingness, phylogenetic clustering and change in mean were important predictors of 
slope error, significant differences in slope error and imputation error.</pre>
<pre>there is no single best solution to deal with missing data
researchers need to assess the available data and consider the need for imputation versus 
limiting the scope of the study or completing analyses for separate groups. Use of data 
imputation should be scrutinized, checking for changes in the data before and after imputation, 
measuring: missingness, phylogenetic clustering, a change in mean and a change in slope</pre>
<pre>The threshold for deciding whether imputation is accurate depends on the research question.</pre>
</div>
<ul>
<li>Currently, at least 160 packages for handling missing data available on the R-CRAN repository (Josse et al., 2020) <span class="math inline">\(\rightarrow\)</span> Testing all available imputation methods was not feasible</li>
<li>imputation can only be successful if it accounts for the mechanism by which data are missing</li>
<li>impact of phylogenetic signal strength on imputation performance already tested (Kim 2018, Molina-Venegas 2018); therefore, standardization of Pagel’s λ between phylogeny and traits at <span class="math inline">\(\sim\)</span> one</li>
</ul>
<p><br/> <br/></p>
</div>
</div>
<div id="synthesis" class="section level1">
<h1>Synthesis</h1>
<p><br/></p>
<div id="methods-description" class="section level2">
<h2>Methods description</h2>
<p><br/></p>
<table>
<colgroup>
<col width="15%" />
<col width="8%" />
<col width="35%" />
<col width="12%" />
<col width="15%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th>METHODS</th>
<th>Group</th>
<th>Precisions</th>
<th>Use phylo</th>
<th>PAPER</th>
<th>SOFTWARE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>mean/median</td>
<td>simple</td>
<td>replace with the mean/median of available trait values</td>
<td>No</td>
<td></td>
</tr>
<tr class="even">
<td>kNN</td>
<td>simple</td>
<td>k-Nearest Neighbour</td>
<td>(eigenvectors from PCoA)</td>
<td>Troyanskaya 2001</td>
<td><code>VIM</code> pkg</td>
</tr>
<tr class="odd">
<td>random forest</td>
<td>simple</td>
<td></td>
<td>(eigenvectors from PCoA)</td>
<td></td>
<td><code>missForest</code> pkg</td>
</tr>
<tr class="even">
<td>dissimilarity</td>
<td>simple</td>
<td><em>species with the same functional strategy have a similar set of functional traits</em> <br/> Gower distance + threshold</td>
<td>No</td>
<td>Westoby 2002, Diaz 2004, Taugourdeau 2014</td>
<td><code>FD</code> pkg</td>
</tr>
<tr class="odd">
<td>relationship</td>
<td>simple/multiple</td>
<td>regression models <br/> but also : kNN, random forest, matrix factorization...</td>
<td>(eigenvectors from PCoA)</td>
<td>Wrigth et al. 2004, 2006</td>
<td><code>dispeRsal</code> function (Tamme 2014)</td>
</tr>
<tr class="even">
<td>ordinary/regression kriging</td>
<td>simple</td>
<td></td>
<td>No</td>
<td></td>
<td><code>automap</code> pkg</td>
</tr>
<tr class="odd">
<td>(R)phylopars</td>
<td>simple (but associated variance)</td>
<td>likelihood-based approach (<code>Phylopars</code>) using both phylogeny and allometric relationships among traits (phylogenetic variance-covariance matrix)</td>
<td>Yes</td>
<td>Bruggeman 2009, Goolsby 2017</td>
<td><code>Phylopars</code> soft, <code>Rphylopars</code> pkg</td>
</tr>
<tr class="even">
<td>PMF</td>
<td>simple</td>
<td>Probabilistic Matrix Factorization, models a sparse matrix as the scalar product of 2 latent matrices to find a factorization minimizing the error between predicted and observed data</td>
<td>No</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HPMF</td>
<td>simple</td>
<td>Hierarchical Probabilistic Matrix Factorization, coupled with phylogenetic information</td>
<td>(taxonomy)</td>
<td>Shan 2012</td>
<td></td>
</tr>
<tr class="even">
<td>BHPMF</td>
<td>simple (but associated variance)</td>
<td>Bayesian Hierarchical Probabilistic Matrix Factorization, coupled with a Gibbs sampler</td>
<td>(taxonomy)</td>
<td>Schrodt 2015</td>
<td><code>BHMPF</code> pkg</td>
</tr>
<tr class="odd">
<td>MICE</td>
<td>multiple</td>
<td>Multivariate imputation by chained equations</td>
<td>(eigenvectors from PCoA)</td>
<td>Azur 2011</td>
<td><code>mice</code> pkg</td>
</tr>
<tr class="even">
<td>jomo</td>
<td>multiple</td>
<td>JOint MOdelling approach for multiple imputation of multilevel data</td>
<td></td>
<td>Quartagno 2019</td>
<td><code>jomo</code> pkg</td>
</tr>
</tbody>
</table>
<p><br/><br/></p>
</div>
<div id="papers-making-comparisons" class="section level2">
<h2>Papers making comparisons</h2>
<p><br/></p>
<table style="width:99%;">
<colgroup>
<col width="5%" />
<col width="18%" />
<col width="8%" />
<col width="12%" />
<col width="18%" />
<col width="16%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th>PAPERS</th>
<th>Tamme (2014)</th>
<th>Taugourdeau (2014)</th>
<th>Penone (2014)</th>
<th>Schrodt (2015)</th>
<th>Poyatos (2018)</th>
<th>Johnson (2020)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>no fam</strong></td>
<td>102</td>
<td></td>
<td></td>
<td>358</td>
<td>4</td>
<td></td>
</tr>
<tr class="even">
<td><strong>no sp</strong></td>
<td>576</td>
<td>1054</td>
<td>273</td>
<td>14320</td>
<td>13</td>
<td>500 (sim)</td>
</tr>
<tr class="odd">
<td><strong>no traits (resp)</strong></td>
<td>1</td>
<td>9</td>
<td>4</td>
<td>13</td>
<td>5</td>
<td>1</td>
</tr>
<tr class="even">
<td><strong>no traits (used)</strong></td>
<td>5</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>4</td>
</tr>
<tr class="odd">
<td><strong>no methods</strong></td>
<td>+ relationship</td>
<td>+ mean <br/> + median <br/> + dissimilarity <br/> + relationship <br/> + mice</td>
<td>+ kNN <br/> + mice <br/> + missForest <br/> + Phylopars</td>
<td>+ PMF <br/> + BHMPF <br/> + aHMPF</td>
<td>+ mean and species mean <br/> + ordinary and regression kriging <br/> + kNN <br/> + mice</td>
<td>+ BHMPF <br/> + Rphylopars <br/> + mice</td>
</tr>
<tr class="even">
<td><strong>missing data</strong></td>
<td></td>
<td>(sim) [10 prob of deletion <span class="math inline">\(0.01 \rightarrow 0.46\)</span>] x [10 rep]</td>
<td>(sim) [<span class="math inline">\(10\% \rightarrow 80\%\)</span>] x [10 rep]</td>
<td>(obs) mean/trait <span class="math inline">\(\sim 79.9\%\)</span></td>
<td>(sim) [<span class="math inline">\(10\% \rightarrow 80\%\)</span>] x [30 rep]</td>
<td>(sim) [2 resp-trait slopes] x [2 cor levels] x [4 bias types] x [2 bias severity] x [<span class="math inline">\(5\% \rightarrow 80\%\)</span>] x [10 rep]</td>
</tr>
<tr class="odd">
<td><strong>evaluation</strong></td>
<td><span class="math inline">\(R^2\)</span></td>
<td>MRdAE</td>
<td>- NRMSE <br/> - effect of dataset/missing data/method on errors <br/> - slope deviation</td>
<td>- RMSE <br/> - <span class="math inline">\(R^2\)</span> <br/> - SMA regression <br/> - procrustes analysis</td>
<td>- NRMSE, KGE <br/> - correlation matrices deviation</td>
<td>- RMSE <br/> - slope deviation</td>
</tr>
<tr class="even">
<td><strong>software</strong></td>
<td><code>dispeRsal</code> function</td>
<td>2 author's functions, <code>mice</code> pkg</td>
<td><code>VIM</code>, <code>mice</code>, <code>missForest</code> pkg, <code>Phylopars</code> soft</td>
<td><code>MATLAB</code></td>
<td><code>automap</code>, <code>VIM</code>, <code>mice</code> pkg</td>
<td><code>BHMPF</code>, <code>Rphylopars</code>, <code>mice</code> pkg</td>
</tr>
</tbody>
</table>
<p><br/><br/></p>
</div>
<div id="methods---according-to-papers" class="section level2">
<h2>Methods +/- (according to papers)</h2>
<p><br/></p>
<table>
<colgroup>
<col width="13%" />
<col width="43%" />
<col width="43%" />
</colgroup>
<thead>
<tr class="header">
<th>METHODS</th>
<th>Positive points</th>
<th>Negative points</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>mean/median</strong></td>
<td>+ accurate for datasets with small percentages of missing values (Schafer 1999)</td>
<td>- least accurate (Taugourdeau 2014)</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>- ignore the variance of the imputed variables (Blonder 2016)</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>- severely altered trait distributions, introduced larger errors in selected trait correlations, tended to cause larger deviations in the correlation matrix (Poyatos 2018)</td>
</tr>
<tr class="even">
<td><strong>kNN</strong></td>
<td>+ can deal with categorical variables (nominal or ordinal) (Penone 2014)</td>
<td>- produced larger errors and induced more bias in the allometric relationship <br/> - must specify a value of the tuning parameter k which is difficult to determine a priori (Penone 2014)</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>- tends to introduce larger bias in bivariate trait relationships compared to MICE (Poyatos 2018)</td>
</tr>
<tr class="even">
<td><strong>random forest</strong></td>
<td>+ performed better without including phylogeny <br/> + can deal with categorical variables (nominal or ordinal) (Penone 2014)</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>dissimilarity</strong></td>
<td>+ Gower dissimilarity can be computed with missing data</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td>+ most accurate when the trait distribution is unbalanced (Taugourdeau 2014)</td>
<td>- cannot be calculated between two species if no trait is documented for both species (Taugourdeau 2014)</td>
</tr>
<tr class="odd">
<td><strong>ordinary/regression kriging</strong></td>
<td></td>
<td>- altered distributions and trait correlations more than mice but they performed similarly in terms of delta cormat at all missingness levels (Poyatos 2018)</td>
</tr>
<tr class="even">
<td><strong>relationship</strong></td>
<td></td>
<td>- depending on the set of species used and traits available, different traits are important (Tamme 2014)</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>- requires several traits per plant/species to be documented <br/> - in most case less accurate than the dissimilarity method <br/> - does not perform well on very unbalanced traits (like SNP) because the multilinear model is strongly governed by extreme values <br/> - very sensitive to the percentage of missing data (Taugourdeau 2014)</td>
</tr>
<tr class="even">
<td><strong>(R)phylopars</strong></td>
<td>+ uses a phylogeny and a sparse trait matrix to estimate simultaneously the across-species (phylogenetic) and within-species (phenotypic) trait covariance (similar to a phylogenetic mixed model) to reconstruct the ancestral state and impute missing values (Goolsby 2017)</td>
<td>- depends on the phylogenetic signal in a trait <br/> - cannot deal with categorical traits</td>
</tr>
<tr class="odd">
<td><strong>PMF</strong></td>
<td></td>
<td>- like PCA, efficient if the original matrix is of low rank, i.e. if the axes of the original matrix provide strong correlations <br/> - accuracy worse than using species mean trait values to fill the gaps (Schrodt 2015)</td>
</tr>
<tr class="even">
<td><strong>HPMF</strong></td>
<td>+ satisfactory to predict trait values when information at the genus level is available (Shan 2012) <br/> + needs only at least one trait value per plant</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>BHPMF</strong></td>
<td>+ On average, across all traits, outperforms PMF, MEAN and aHPMF, with MEAN being significantly more accurate than PMF <br/> + advantage over MEAN largest for ‘physiological traits’(leaf N, leaf P), and smaller for more ‘structural traits’ (seed mass, plant height) <br/> + BHPMF and MEAN capture these general trait–trait correlations, but BHPMF reproduces extreme values more accurately than MEAN and is therefore generally better at capturing the shape of the scatter of observed trait data (Schrodt 2015)</td>
<td>- presence of strong trait–trait correlations is a prerequisite for the accuracy of BHPMF (Schrodt 2015)</td>
</tr>
<tr class="even">
<td><strong>MICE</strong></td>
<td>+ more accurate than all other methods for all traits except for the specific leaf area (Taugourdeau 2014)</td>
<td>- affected by the percentage of missing data for 6 (whole subdatabase) and 7 (herbaceous subdatabase) traits (Taugourdeau 2014)</td>
</tr>
<tr class="odd">
<td></td>
<td>+ smaller error and bias as compared to other multiple imputation approaches (Ambler, Omar &amp; Royston 2007)</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td>+ performed better without including phylogeny <br/> + can deal with categorical variables (nominal or ordinal) (Penone 2014)</td>
<td>- linear dependencies between variables cause fatal errors and should be eliminated before imputation (Penone 2014)</td>
</tr>
<tr class="odd">
<td></td>
<td>+ closely tracked observed trait distributions, introduced the least error in trait correlations under high missingness levels and yielded low delta cormat at extreme missingness levels (Poyatos 2018) <br/> + perform well when there is no true relationship between response and trait (Johnson 2020)</td>
<td>- perform poorly when there is a positive relationship between response and trait, even when including the response <br/> - phylogenetic information should only be included when data are missing with no bias or a weak bias(Johnson 2020)</td>
</tr>
<tr class="even">
<td><strong>jomo</strong></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p><br/> <br/></p>
</div>
</div>
<div id="citations" class="section level1">
<h1>Citations</h1>
<!-- - Audigier V, White I R, Jolani S, Debray T PA, Quartagno M, Carpenter J, van Buuren S and Resche-Rigon M. 2018. “Multiple Imputation for Multilevel Data with Continuous and Binary Variables.” Statistical Science 33 (2): 160–83. https://doi.org/10.1214/18-STS646 -->
<ul>
<li>Azur, M J, Stuart E A, Frangakis C and Leaf P J. 2011. “Multiple Imputation by Chained Equations: What Is It and How Does It Work?” International Journal of Methods in Psychiatric Research 20 (1): 40–49. <a href="https://doi.org/10.1002/mpr.329" class="uri">https://doi.org/10.1002/mpr.329</a> <!-- - Blonder, Benjamin. 2016. “Do Hypervolumes Have Holes?” American Naturalist 187 (4): E93–105. https://doi.org/10.1086/685444 --></li>
<li>Diniz-Filho, J A F, Villalobos F, and Bini L M. 2015. “The Best of Both Worlds: Phylogenetic Eigenvector Regression and Mapping.” Genetics and Molecular Biology 38 (3): 396–400.</li>
<li>Goolsby, E W, Bruggeman J and Ané C. 2017. “Rphylopars: Fast Multivariate Phylogenetic Comparative Methods for Missing Data and within-Species Variation.” Methods in Ecology and Evolution 8 (1): 22–27. <a href="https://doi.org/10.1111/2041-210X.12612" class="uri">https://doi.org/10.1111/2041-210X.12612</a> <!-- - Huque Md H, Carlin J B, Simpson J A and Lee K J. 2018. “A Comparison of Multiple Imputation Methods for Missing Data in Longitudinal Studies 01 Mathematical Sciences.” BMC Medical Research Methodology 18 (1): 1–16. https://doi.org/10.1186/s12874-018-0615-6 --></li>
<li>Johnson T F, Isaac N J B, Paviolo A and González‐Suárez M. 2020. “Handling Missing Values in Trait Data.” Global Ecology and Biogeography, no. December 2019: 1–12. <a href="https://doi.org/10.1111/geb.13185" class="uri">https://doi.org/10.1111/geb.13185</a> <!-- - Kim S W, Blomberg S P and Pandolfi J M. 2018. “Transcending Data Gaps: A Framework to Reduce Inferential Errors in Ecological Analyses.” Ecology Letters. https://doi.org/10.1111/ele.13089 --></li>
<li>Molina-Venegas R, Moreno-Saiz J C, Parga I C, Davies T J, Peres-Neto P R and Rodríguez M. 2018. “Assessing Among-Lineage Variability in Phylogenetic Imputation of Functional Trait Datasets.” Ecography 41 (10): 1740–49. <a href="https://doi.org/10.1111/ecog.03480" class="uri">https://doi.org/10.1111/ecog.03480</a></li>
<li>Penone C, Davidson A D, Shoemaker K T, Di Marco M, Rondinini C, Brooks T M, Young B E, Graham C H and Costa G C. 2014. “Imputation of Missing Data in Life-History Trait Datasets: Which Approach Performs the Best?” Methods in Ecology and Evolution 5 (9): 961–70. <a href="https://doi.org/10.1111/2041-210X.12232" class="uri">https://doi.org/10.1111/2041-210X.12232</a></li>
<li>Poyatos R, Sus O, Badiella L, Mencuccini M and Martínez-Vilalta J. 2018. “Gap-Filling a Spatially Explicit Plant Trait Database: Comparing Imputation Methods and Different Levels of Environmental Information.” Biogeosciences 15 (9): 2601–17. <a href="https://doi.org/10.5194/bg-15-2601-2018" class="uri">https://doi.org/10.5194/bg-15-2601-2018</a></li>
<li>Quartagno M, Grund S and Carpenter J. 2019. “Jomo: A Flexible Package for Two-Level Joint Modelling Multiple Imputation.” R Journal 11 (2): 205–28. <a href="https://doi.org/10.32614/RJ-2019-028" class="uri">https://doi.org/10.32614/RJ-2019-028</a></li>
<li>Schrodt F, Kattge J, Shan H, Fazayeli F, Joswig J, Banerjee A, Reichstein M, et al. 2015. “BHPMF - a Hierarchical Bayesian Approach to Gap-Filling and Trait Prediction for Macroecology and Functional Biogeography.” Global Ecology and Biogeography 24 (12): 1510–21. <a href="https://doi.org/10.1111/geb.12335" class="uri">https://doi.org/10.1111/geb.12335</a></li>
<li>Shan H, Kattge J, Reich P B, Banerjee A, Schrodt F and Reichstein M. 2012. “Gap Filling in the Plant Kingdom - Trait Prediction Using Hierarchical Probabilistic Matrix Factorization.” In Proceedings of the 29th International Conference on Machine Learning, ICML 2012, 2:1303–10.</li>
<li>Tamme R, Götzenberger L, Zobel M, Bullock J M, Hooftman D AP, Kaasik A and Pärtel M. 2014. “Predicting Species’ Maximum Dispersal Distances from Simple Plant Traits.” Ecology 95 (2): 505–13. <a href="https://doi.org/10.1890/13-1000.1" class="uri">https://doi.org/10.1890/13-1000.1</a></li>
<li>Taugourdeau S, Villerd J, Plantureux S, Huguenin-Elie O and Amiaud B. 2014. “Filling the Gap in Functional Trait Databases: Use of Ecological Hypotheses to Replace Missing Data.” Ecology and Evolution 4 (7): 944–58. <a href="https://doi.org/10.1002/ece3.989" class="uri">https://doi.org/10.1002/ece3.989</a></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
